app/models/user.rb

```diff
@@ -21,4 +33,23 @@ def name
   def name=(name)
     self.first_name, self.last_name = name.split(' ')
   end
+
+  def approved!
+    errors = []
+    self.approved = true
+    if self.save
+      thread do
+        UserMailer.send_approval(self)
```

Comment:
Any specific reason for using native threads here instead of abstraction provided by deliver_later?

Reply:
Considered more items to do in background, not only sending email.